python/lancedb/table.py

```diff
@@ -159,10 +174,12 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         return self.__repr__()
 
+    @_not_supported_on_remote
     def head(self, n=5) -> pa.Table:
         """Return the first n rows of the table."""
         return self._dataset.head(n)
 
+    @_not_supported_on_remote
```

Comment:
why is `to_pandas()/to_arrow()` not supported?

Reply:
these two are fetching the entire table. 

we don't have an API for this yet.